### PR TITLE
[build] Clean up prj.conf and other config files in build

### PR DESCRIPTION
**Pull request by brianjjones**
This is also a fix for #826

Signed-off-by: Brian J Jones <brian.j.jones@intel.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -171,7 +171,7 @@ analyze: $(JS)
 		echo "ccflags-y += -DZJS_SNAPSHOT_BUILD" >> src/Makefile; \
 	fi
 ifeq ($(DEV), ashell)
-	@cat fragments/prj.mdef.dev >> prj.mdef
+	@cat fragments/prj.mdef.ashell >> prj.mdef
```
Comment (jimmy-huang) [inline Makefile]: couldn't this be moved into the bottom section where you append the prj.conf? so you don't need a duplicate ifeq

File: fragments/prj.conf.ashell [hunk 2]
```diff
@@ -1,14 +1,10 @@
-# HEAP
-CONFIG_HEAP_MEM_POOL_SIZE=4096
-
 # SHELL
 CONFIG_EARLY_CONSOLE=y
```
Comment (jimmy-huang) [inline fragments/prj.conf.ashell]: if you are already get these configs from prj.conf.base, then there's a lot of duplicate here like STDOUT_CONSOLE, FLOAT, NEWLIB

**Comment by grgustaf** [comment]
+1

File: Makefile (hunk 1 above)
Comment (grgustaf) [inline Makefile]: They're in different sections, I don't want to risk that right now. The mdef stuff is in one place, the prj.conf is in one place.

**Pull request merged**